doc/howtos/OSTree.rst

```diff
@@ -0,0 +1,153 @@
+Handling System Updates Using OSTree
+####################################
+
+IoT Reference OS Kit ('refkit') has support for updating devices running
+one of the refkit profile images using an HTTP/HTTPs server and OSTree.
+OSTree is a library and suite of command line tools, largely inspired by
+git, that provides version control and a distribution mechanism for bootable
+OS filesystem trees, or other binaries. For a comprehensive introduction
+to OSTree, an overview of its architecture and feature set, please refer
+to its `documentation <http://ostree.readthedocs.io>_`.
+
+If enabled, OSTree support in refkit
+
+ * provides A/B-versioning (within a single block device partition)
+ * creates a per-image OSTree repository for each image built
+ * exposes builds in a common OSTree repository as a series of commits
+ * can sign each commit (which is then verified during updates)
+ * can provide a service for automatic image updates
+
+
+Enabling OSTree Support
+#######################
+
+To enable end-to-end OSTree support with automatic image updates, you will
+need to
+
+ * enable OSTree support for your builds
+ * ideally generate and use a pair of signing keys for your builds/updates
+ * expose the OSTree repository of your builds over HTTP/HTTPS
+ * point your clients to the exposed OSTree repository
+
+To enable OSTree support, turn on the 'ostree' image feature. You can
+do this by incuding the following configuration snippet in your local.conf
+or other suitable global Yocto configuration file::
+
+    REFKIT_IMAGE_EXTRA_FEATURES += "ostree"
+
+You can generate a pair of signing keys with the following command::
```

Comment:
I still think we should mention here that generating keys is not needed for "development" images.